src/webgpu/shader/execution/memory_model/memory_model_setup.ts

```diff
@@ -249,52 +272,60 @@ export class MemoryModelTester {
       size: shuffledWorkgroupsSize,
     };
 
+    // Combine 3 arrays into 1 buffer as we need to keep the number of storage buffers to 4 for compat.
```

Comment:
The three logical arrays are jammed up right beside each other.

This induces false sharing between the combo.barrier.value[0] location and the  scratchpad beside it.
These tests were developed as part of research on how to heuristically evoke weak behaviours to more effectively test memory models.  https://dl.acm.org/doi/10.1145/3575693.3575750 
So I'd like to avoid adding false sharing as that would likely perturb the results in a meaningful way.

Please quantize the sizes so they are unlikely to have overlapping cache lines.
E.g.    
```js
const falseSharingAvoidanceQuantum = 4096;
const barrierSize = Math.max(bytesPerWord, falseSharingAvoidanceQuantum);
const scratchpadSize = Math.max(this.params.scratchMemorySize * bytesPerWord, falseSharingAvoidanceQuantum);
const scratchMemoryLocationsSize = Math.max(this.params.maxWorkgroups * bytesPerWord, falseSharingAvoidanceQuantum);
```

Reply:
Is `max` what you wanted here or should it be `align` as in

const barrierSize = align(bytesPerWord, falseSharingAvoidanceQuantum);
const scratchpadSize = align(this.params.scratchMemorySize * bytesPerWord, falseSharingAvoidanceQuantum);
const scratchMemoryLocationsSize = align(this.params.maxWorkgroups * bytesPerWord, falseSharingAvoidanceQuantum);

I put `max` in for now.

Reply:
You're right align would be better.  I didn't look hard enough at `util/math.ts`.   :-)